hawkbit/src/ddi/deployment_base.rs

```diff
@@ -284,6 +285,15 @@ impl<'a> Chunk<'a> {
             .map(move |a| Artifact::new(a, client.clone()))
     }
 
+    /// An iterator on all the metadata of the chunk.
```

Comment:
Doc mentions an iterator but you are actually returning a `HashMap`. I think an iterator would make the more sense here.  Would this fit your needs?

```rust
pub fn metadata(&self) -> impl Iterator<Item = (&str, &str)> {
        self.chunk
            .metadata
            .iter()
            .map(|a| (a.key.as_str(), a.value.as_str()))
    }
```

Reply:
That also works for me. Changed it.